SMLHelper/FishFramework/Extensions.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using UnityEngine;
+
+namespace SMLHelper.V2.FishFramework
+{
+    public static class Extensions
+    {
+        public static T AddOrGet<T>(this GameObject obj) where T : Component
```

Comment:
This could probably be implemented in Utilities so other stuff can use it. Also I would suggest renaming it to GetOrAddComponent, since it gets before it adds, and it's about components